nixos/modules/services/development/mlflow/default.nix

```diff
@@ -0,0 +1,132 @@
+{ options, config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.mlflow;
+
+  keyToFlag = k: (if lib.stringLength k == 1 then "-" else "--") + k;
+
+  toFlags = k: v:
+    if v == false || v == null then
+      ""
+    else if v == true then
+      keyToFlag k
+    else if lib.isList v then
+      lib.concatMapStringsSep " " (x: toFlags k x) v
+    else
+      "${keyToFlag k} ${toString v}";
+
+  flags = lib.concatStringsSep " " (lib.mapAttrsToList toFlags cfg.settings);
+
+  valueType = with lib.types;
+    nullOr (oneOf [ bool float int package path port str ]);
+
+  settingsType = with lib.types;
+    submodule {
+      freeformType = attrsOf (oneOf [ valueType (listOf valueType) ]);
+      options = {
+        host = lib.mkOption {
+          type = str;
+          default = "127.0.0.1";
+          description = ''
+            The IP address the mlflow server should listen on.
+          '';
+        };
+
+        port = lib.mkOption {
+          type = port;
+          default = 5000;
+          description = ''
+            The TCP port the mlflow server should listen on.
+          '';
+        };
+
+        default-artifact-root = lib.mkOption {
+          type = str;
+          default = if cfg.settings.serve-artifacts or false then
+            "mlflow-artifacts:/"
+          else
+            "/var/lib/mlflow/mlruns";
+
+          defaultText = lib.literalExpression ''
+            if cfg.settings.serve-artifacts or false then
+              "mlflow-artifacts:/"
+            else
+              "/var/lib/mlflow/mlruns"
+          '';
+
+          description = ''
+            Directory in which to store artifacts for any new experiments
+            created. For tracking server backends that rely on SQL, this option
+            is required in order to store artifacts.
+          '';
+        };
+
+        backend-store-uri = lib.mkOption {
+          type = str;
+          default = "sqlite:////var/lib/mlflow/mlruns.db";
+          description = ''
+            URI to which to persist experiment and run data.
+
+            Acceptable URIs are SQLAlchemy-compatible database connection
+            strings (e.g. 'sqlite:///path/to/file.db') or local filesystem URIs
+            (e.g. 'file:///absolute/path/to/directory'). Using a SQL-backed
+            store is required for a number of features.
```

Comment:
Postgres is supported, as are any other SQL engines supported by SQLAlchemy so it's kind of implied by referring to SQLAlchemy. I chose SQLite as default because it requires no extra services running and no configuration for authentication etc... I could mention Postgres explicitly if you want.

Reply:
I think it would be great to have a `postgresql:///mlflow` example for people who run in trust setups with a `mlflow` database.

Reply:
tbh, I don't know how to do that, the sqlalchemy documentation only gives examples using a password. But postgresql requires extra configuration and extra care around security while a sqlite database does not, which is why I'd much rather recommend a sqlite database. Also note that this is just copy-pasted from the output of `mlflow server --help` so I guess it's an upstream recommendation.